ckanext/archiver/tasks.py

```diff
@@ -548,11 +548,54 @@ def archive_resource(context, resource, log, result=None, url_timeout=30):
         log.warning('Not saved cache_url because no value for '
                     'ckanext.archiver.cache_url_root in config')
         raise ArchiveError(_('No value for ckanext.archiver.cache_url_root in config'))
-    cache_url = urlparse.urljoin(context['cache_url_root'],
-                                 '%s/%s' % (relative_archive_path, file_name))
+
+    archiver_s3upload_enable = config.get('ckanext.archiver.s3upload_enable')
+    resource_id_dir = relative_archive_path.split('/')[-1]
+
+    if archiver_s3upload_enable:
+        upload_obj, key_path = upload_archived_resource(resource_id_dir, file_name, saved_file)
+        cache_url =  generate_cache_url(upload_obj, key_path)
+    else:
+        cache_url = urlparse.urljoin(context['cache_url_root'],
+                                    '%s/%s' % (relative_archive_path, file_name))
+
     return {'cache_filepath': saved_file,
             'cache_url': cache_url}
 
+def upload_archived_resource(resource_id_dir, filename, saved_file):
+    '''
+    Uploads the resources to s3filestore in directory
+    <S3FILESTORE__AWS_BUCKET_NAME>/<S3FILESTORE__AWS_STORAGE_PATH>/archived_resources/
+    '''
+    storage_path = config.get('ckanext.s3filestore.aws_storage_path')
```

Comment:
Seems like you are depended on `aws_storage_path`. Let's not just assume that it's set but a validation error if it's not. Or is it not required? If not required this will return `None` and `os.path.join` will error. in that case do `config.get('ckanext..', '')` so that you get empty string

Reply:
It is required and is usually available for all the instances as it is used while resources are uploaded to a filestore by instances.
Will add validation error if not available in config.

Reply:
Let's document. This extension is used outside of our instances as well

Reply:
Will add relevant info in the README

for storage path we can do something like

```
if not config.get('ckanext.s3filestore.aws_storage_path'):
        log.warning('Not saved to filestore because no value for '
                    'ckanext.s3filestore.aws_storage_path in config')
        raise ArchiveError(_('No value for ckanext.s3filestore.aws_storage_path in config'))
```

Reply:
Implemented the above-mentioned logic

Added config settings. SEE Point no. 6 of README.
https://github.com/datopian/ckanext-archiver/blob/feature/filestore_resource_upload/README.rst#config-settings